lib/util/http/github.ts

```diff
@@ -301,20 +302,36 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
     const result: T[] = [];
 
     const { paginate = true } = options;
-    let count = options.count || 100;
+
+    const initialCount = options.count || 100;
+    let count = initialCount;
     let limit = options.limit || 1000;
     let cursor: string = null;
 
     let isIterating = true;
     while (isIterating) {
+      const actualCount = Math.min(count, limit);
       const res = await this.requestGraphql<GithubGraphqlRepoData<T>>(query, {
         ...options,
-        count: Math.min(count, limit),
+        count: actualCount,
         cursor,
         paginate,
       });
       const fieldData = res?.data?.repository?.[fieldName];
       if (fieldData) {
+        // See: https://github.com/renovatebot/renovate/issues/12703
+        // istanbul ignore if
+        if (
+          count < initialCount &&
+          count === actualCount &&
```

Comment:
So this will log for each page, only if the number returned is equal to the maximum requested? e.g. if we halve from 100 to 50, and there's 251 total, then we'd get this log message 5 times? But if we halve from 100 to 50 and there's 49 total, we'd get no log message?

Reply:
Would we be better to add this logic further down once iterating is done, to ensure we log it exactly once if the count was reduced?

Reply:
Changed its logic the way it should log only shrinked result, so iteration `100 -> error -> 50 -> 50 -> 1` should report 50 (not 1) for total 101 items